atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionOutcomeMetricsAssert.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl;
+
+import java.util.function.LongConsumer;
+
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.WritableAssertionInfo;
+import org.assertj.core.internal.Objects;
+
+import com.codahale.metrics.Metric;
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.logging.LoggingArgs;
+import com.palantir.tritium.metrics.registry.MetricName;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+
+public class TransactionOutcomeMetricsAssert extends
+        AbstractAssert<TransactionOutcomeMetricsAssert, TransactionOutcomeMetrics> {
+    private final TaggedMetricRegistry taggedMetricRegistry;
+    private final Objects objects = Objects.instance();
+    private WritableAssertionInfo writableAssertionInfo = new WritableAssertionInfo();
+
+    public TransactionOutcomeMetricsAssert(TransactionOutcomeMetrics actual) {
+        super(actual, TransactionOutcomeMetricsAssert.class);
+        taggedMetricRegistry = actual.metricsManager.getTaggedRegistry();
+    }
+
+    public static TransactionOutcomeMetricsAssert assertThat(TransactionOutcomeMetrics actual) {
+        return new TransactionOutcomeMetricsAssert(actual);
+    }
+
+    public TransactionOutcomeMetricsAssert hasSuccessfulCommits(long count) {
+        checkPresentAndCheckCount("successfulCommit", count);
```

Comment:
`code.refactor.stability.Optional`: Consider making the outcomes into an enum, which would allow you to reference the string here rather than duplicating it.

Reply:
Nice package naming trend 😄 This one is worth doing. I initially thought about this but didn't push on the basis that a tx can cause multiple outcomes, but on revisiting it that argument doesn't really make sense.